### PR TITLE
new: working on conference or booth object and event type #FoundationFunding

### PR DESCRIPTION
**Pull request by FHenry**
class conference or booth or and table actioncomm ok

### CODE REVIEW

File: htdocs/comm/action/class/actioncomm.class.php [hunk 1]
```diff
@@ -539,14 +557,23 @@ public function create(User $user, $notrigger = 0)
...
+		$sql .= (!empty($this->event_paid) ? (int) $this->event_paid : 0).", ";
+		$sql .= (!empty($this->status) ? (int) $this->status : "null");
 		$sql .= ")";
 
 		dol_syslog(get_class($this)."::add", LOG_DEBUG);
 		$resql = $this->db->query($sql);
 		if ($resql) {
 			$this->ref = $this->id = $this->db->last_insert_id(MAIN_DB_PREFIX."actioncomm", "id");
-
+			$sql = "UPDATE ".MAIN_DB_PREFIX."actioncomm SET ref='".$this->ref."' WHERE id=".$this->id ;
```
Comment (frederic34) [inline htdocs/comm/action/class/actioncomm.class.php:570]: ref not escaped

File: htdocs/comm/action/class/actioncomm.class.php [hunk 2]
```diff
@@ -717,13 +744,14 @@ public function fetch($id, $ref = '', $ref_ext = '', $email_msgid = '')
 		$sql .= " c.id as type_id, c.code as type_code, c.libelle as type_label, c.color as type_color, c.picto as type_picto,";
 		$sql .= " s.nom as socname,";
 		$sql .= " u.firstname, u.lastname as lastname";
```
Comment (frederic34) [inline htdocs/comm/action/class/actioncomm.class.php:746]: missing comma after lastname

Comment (FHenry) [reply htdocs/comm/action/class/actioncomm.class.php:746]: Yep fix

File: htdocs/comm/action/class/actioncomm.class.php [hunk 3]
```diff
@@ -1760,6 +1801,7 @@ public function build_exportfile($format, $type, $cachedelay, $filename, $filter
 			$sql .= " u.firstname, u.lastname, u.email,";
 			$sql .= " s.nom as socname,";
 			$sql .= " c.id as type_id, c.code as type_code, c.libelle as type_label";
```
Comment (frederic34) [inline htdocs/comm/action/class/actioncomm.class.php:1803]: missing comma after type_label

File: htdocs/comm/action/class/actioncomm.class.php [hunk 4]
```diff
@@ -716,14 +743,15 @@ public function fetch($id, $ref = '', $ref_ext = '', $email_msgid = '')
 		$sql .= " a.email_msgid, a.email_subject, a.email_from, a.email_to, a.email_tocc, a.email_tobcc, a.errors_to,";
 		$sql .= " c.id as type_id, c.code as type_code, c.libelle as type_label, c.color as type_color, c.picto as type_picto,";
 		$sql .= " s.nom as socname,";
-		$sql .= " u.firstname, u.lastname as lastname";
+		$sql .= " u.firstname, u.lastname as lastname,";
+		$sql .= " num_vote, event_paid, status";
```
Comment (frederic34) [inline htdocs/comm/action/class/actioncomm.class.php:747]: status is ambiguous, add c. u. or s.

File: htdocs/comm/action/class/actioncomm.class.php [hunk 5]
```diff
@@ -1760,6 +1801,7 @@ public function build_exportfile($format, $type, $cachedelay, $filename, $filter
 			$sql .= " u.firstname, u.lastname, u.email,";
 			$sql .= " s.nom as socname,";
 			$sql .= " c.id as type_id, c.code as type_code, c.libelle as type_label";
+			$sql .= " num_vote, event_paid, status";
```
Comment (frederic34) [inline htdocs/comm/action/class/actioncomm.class.php:1804]: also here c. u. or s.

File: htdocs/comm/action/class/actioncomm.class.php (hunk 4 above)
Comment (FHenry) [reply htdocs/comm/action/class/actioncomm.class.php:747]: yes true thanks

File: htdocs/comm/action/class/actioncomm.class.php (hunk 5 above)
Comment (FHenry) [reply htdocs/comm/action/class/actioncomm.class.php:1804]: yes true thanks

**Comment by frederic34** [comment]
To add ref in adherent I did that (because I seen some external module which have already added filed)
https://github.com/Dolibarr/dolibarr/blob/43e58f70a152202534d56daaf70a4f348832ce8b/htdocs/install/mysql/migration/13.0.0-14.0.0.sql#L53-L56
perhaps you can do similar thing

**Pull request merged**